pkg/prometheus/promcfg.go

```diff
@@ -1249,9 +1257,13 @@ func (cg *ConfigGenerator) generateServiceMonitorConfig(
 
 	// Filter targets based on correct port for the endpoint.
 	if ep.Port != "" {
+		sourceLabels := []string{"__meta_kubernetes_endpoint_port_name"}
```

Comment:
this will likely need to be addressed in the *deprecated* `TargetPort` block below also

Reply:
The other block refers to `__meta_kubernetes_pod_container_port_name` so I'm not sure it needs an update?

Reply:
yes, you are right, sorry